subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ArtifactViewAttributesIntegrationTest.groovy

```diff
@@ -0,0 +1,392 @@
+            $artifactComparison
+        """
+
+        expect:
+        run ':help'
+    }
+
+
+    def "declare and iterate incoming files, then incoming artifacts, then artifact view files, then artifact view artifacts"() {
+        buildFile << """
+            $declareIncomingFiles
+            $iterateIncomingFiles
+
+            $declareIncomingArtifacts
+            $iterateIncomingArtifacts
+
+            $declareArtifactViewFiles
+            $iterateArtifactViewFiles
+
+            $declareArtifactViewArtifacts
+            $iterateArtifactViewArtifacts
+
+            $filesComparison
+            $artifactComparison
+        """
+
+        expect:
+        run ':help'
+    }
+
+    def "use legacy behavior to declare and iterate artifact view files, then declare and iterate incoming files"() {
+        System.setProperty(DefaultConfiguration.USE_LEGACY_ATTRIBUTE_SNAPSHOT_BEHAVIOR, Boolean.TRUE.toString())
+
+        buildFile << """
+            $declareArtifactViewFiles
+            $iterateArtifactViewFiles
+
+            $declareIncomingFiles
+            $iterateIncomingFiles
+
+            // The legacy behavior was for the artifact view variant selection to not use the lazy attributes
+            assert incomingFiles*.name != artifactViewFiles*.name
+        """
+
+        expect:
+        executer.expectDocumentedDeprecationWarning("The org.gradle.configuration.use-legacy-attribute-snapshot-behavior system property has been deprecated. This is scheduled to be removed in Gradle 9.0. Please remove this flag and use the current default behavior. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#legacy_attribute_snapshotting")
+        run ':help'
+    }
+
+    def "use legacy behavior to declare and iterate incoming artifacts, then declare and iterate artifact view artifacts"() {
+        System.setProperty(DefaultConfiguration.USE_LEGACY_ATTRIBUTE_SNAPSHOT_BEHAVIOR, Boolean.TRUE.toString())
+
+        buildFile << """
+            $declareIncomingArtifacts
+            $iterateIncomingArtifacts
+
+            $declareArtifactViewArtifacts
+            $iterateArtifactViewArtifacts
+
+            // The legacy behavior was for the incoming artifacts variant selection to not use the lazy attributes
+            assert incomingArtifacts*.id.file.name != artifactViewArtifacts*.id.file.name
+        """
+
+        expect:
+        executer.expectDocumentedDeprecationWarning("The org.gradle.configuration.use-legacy-attribute-snapshot-behavior system property has been deprecated. This is scheduled to be removed in Gradle 9.0. Please remove this flag and use the current default behavior. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#legacy_attribute_snapshotting")
+        run ':help'
+    }
+
+    @Ignore("There are 10,000s of these, so it takes some time, but is the ultimate sanity check")
```

Comment:
I feel that if this test is never going to run it only adds to clutter and adds to code which we need to maintain. I think that after sanity has indeed been checked this is no longer necessary. IMO the other tests in this class should give us the confidence such that this is not necessary to include here.

Reply:
That's fine, it lives in history.